src/common/detectWMDE.c

```diff
@@ -275,6 +278,28 @@ static void getXFCE4(FFinstance* instance, FFWMDEResult* result)
     }
 }
 
+static void getLXQt(FFinstance* instance, FFWMDEResult* result)
+{
+    ffStrbufSetS(&result->deProcessName, "lxqt-session");
+    ffStrbufSetS(&result->dePrettyName, "LXQt");
+    ffParsePropFile("/usr/share/cmake/lxqt/lxqt-config.cmake", "set(LXQT_VERSION", &result->deVersion);
+
+    if(result->deVersion.length == 0 && instance->config.allowSlowOperations)
+    {
+        //This is really, really, really slow. Thank you, LXQt developers
+        ffProcessAppendStdOut(&result->deVersion, (char* const[]){
+            "lxqt-session",
+            "-v",
+            NULL
+        });
+
+        FF_STRBUF_CREATE(tmp);
+        ffGetPropValueFromLines(result->deVersion.chars , "liblxqt", &tmp);
+        ffStrbufSet(&result->deVersion, &tmp);
+        ffStrbufDestroy(&tmp);
```

Comment:
This is a unnecesarry buffer allocation. Since we will always copy back to front, in an index increasing order, we can safely use only one buffer. We set the length to 0, so appending will start at index 0, but leave the null byte at the old index, because `ffgetPropValue*` functions use it to detect the length. Implementation would look something like this:
```c
result->deVersion.length = 0; //don't set '\0' byte
ffGetPropValueFromLines(result->deVersion.chars , "liblxqt", &result->deVersion);
```
I didn't test it, but it _should_ work.

Reply:
It worked. :+1: 
I tried it before without the length adjustment, which gave me all the lines, so I used a tmp buffer instead.

Reply:
Yes this is poor method naming on my side. `ffGetPropValue*` functions actually append the buffer instead of setting it.